packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -394,20 +394,21 @@ export default util.createRule<Options, MessageIds>({
           },
         },
 
-      'TSPropertySignature[computed = false]': {
-        validator: validators.typeProperty,
-        handler: (
-          node: TSESTree.TSPropertySignatureNonComputedName,
-          validator,
-        ): void => {
-          const modifiers = new Set<Modifiers>([Modifiers.public]);
-          if (node.readonly) {
-            modifiers.add(Modifiers.readonly);
-          }
+      'TSPropertySignature[computed = false][typeAnnotation.typeAnnotation.type != "TSFunctionType"]':
```

Comment:
I'm wondering if this `typeAnnotation.typeAnnotation.type` part should be moved to some variable. It is later on used in L466 in `typeMethod` selector

Reply:
You know, we've been leaning away from using these fancier queries, and instead towards simpler, more readable ones. https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/339

Not requesting changes here as we haven't taken action on that yet & this file already has a lot of complex queries. Just food for thought. 😄